File: GuardianMonsters/inventory/src/main/java/de/limbusdev/guardianmonsters/inventory/AInventorySubMenu.java
package main.java.de.limbusdev.guardianmonsters.inventory;

import com.badlogic.gdx.scenes.scene2d.Group;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.utils.Align;

import java.util.Observable;
import java.util.Observer;

import de.limbusdev.guardianmonsters.ui.Constant;


/**
 * @author Georg Eckert 2017
 */

public abstract class AInventorySubMenu extends Group {

    private Skin skin;
    public static final int TOOLBAR_HEIGHT = 36;
    private InventoryScreen core;

    public AInventorySubMenu(Skin skin) {
        super();
        this.skin = skin;
        setDebug(Constant.DEBUGGING_ON, true);
        setSize(Constant.WIDTH, Constant.HEIGHT-TOOLBAR_HEIGHT);
        setPosition(0,0, Align.bottomLeft);
    }

    public InventoryScreen getCore() {
        return core;
    }

    public void setCore(InventoryScreen core) {
        this.core = core;
    }

    public boolean hasCore() {

        return (core != null);
    }

    public Skin getSkin() {
        return skin;
    }

    public abstract void refresh();
}

File: GuardianMonsters/inventory/src/de/limbusdev/guardianmonsters/inventory/AInventorySubMenu.kt
package de.limbusdev.guardianmonsters.inventory

import com.badlogic.gdx.scenes.scene2d.Group
import com.badlogic.gdx.scenes.scene2d.ui.Skin
import com.badlogic.gdx.utils.Align

import de.limbusdev.guardianmonsters.ui.Constant


/**
 * @author Georg Eckert 2017
 */

abstract class AInventorySubMenu() : Group()
{
    // ............................................................................................. PROPERTIES
    var core: InventoryScreen? = null


    // ............................................................................................. CONSTRUCTORS
    init
    {
        setDebug(Constant.DEBUGGING_ON, true)
        setSize(Constant.WIDTHf, Constant.HEIGHTf - TOOLBAR_HEIGHT)
        setPosition(0f, 0f, Align.bottomLeft)
    }


    // ............................................................................................. METHODS
    /**
     * Used to update the currently selected Guardian of this SubMenu, if necessary.
     * @param teamPosition
     */
    open fun syncSelectedGuardian(teamPosition: Int) {}

    /**
     * Tells the core about the currently selected Guardian
     * @param teamPosition
     */
    protected fun propagateSelectedGuardian(teamPosition: Int)
    {
        core?.setCurrentlyChosenTeamMember(teamPosition)
    }

    abstract fun refresh()

    protected abstract fun layout(skin: Skin)

    companion object
    {
        val TOOLBAR_HEIGHT = 36
    }
}
